backend/capellacollab/sessions/routes.py

```diff
@@ -257,15 +257,15 @@ def models_as_json(
             )
         ):
             continue
-        yield git_model_as_json(git_model, entry.deep_clone)
+        yield git_model_as_json(git_model, entry.revision, entry.deep_clone)
 
 
 def git_model_as_json(
-    git_model: git_models.DatabaseGitModel, deep_clone: bool
+    git_model: git_models.DatabaseGitModel, revision: str, deep_clone: bool
 ) -> dict[str, str | int]:
     d = {
         "url": git_model.path,
-        "revision": git_model.revision,
+        "revision": revision or git_model.revision,
```

Comment:
Either the `or git_model.revision` is unnecessary here or the type hint should of `revision` should be `str | None`. However, from what I can see, we only use the `git_model_as_json` function here and `entry.revision` should never be `None`. Therefore, I would suggest to simply remove the `or git_model.revision`.

Reply:
Indeed, it's probably caught by the model validator anyway.